### ISSUE TITLE
NullPointer at DroppyMenuPopup.dismissPopup

### ISSUE THREAD

**Issue by javieramp** [shehabic/Droppy#42]
Here is the trace:

java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.ViewGroup.removeView(android.view.View)' on a null object reference
at com.shehabic.droppy.DroppyMenuPopup.dismissPopup (DroppyMenuPopup.java:155)
at com.shehabic.droppy.DroppyMenuPopup.dismiss (DroppyMenuPopup.java:150)
at com.shehabic.droppy.DroppyMenuPopup$2.onClick (DroppyMenuPopup.java:105)
- 10 non-project frames
  at android.view.View.performClick (View.java:4757)
  at android.view.View$PerformClick.run (View.java:19757)
  at android.os.Handler.handleCallback (Handler.java:739)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:135)
  at android.app.ActivityThread.main (ActivityThread.java:5219)
  at java.lang.reflect.Method.invoke (Unknown source)
  at java.lang.reflect.Method.invoke (Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:898)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:693)

I think that fail comes at this point in DroppyLibrary:

```
protected void dismissPopup(boolean itemSelected)
{
    ((ViewGroup) mContentView.getParent()).removeView(mContentView);
    ((ViewGroup) modalWindow.getParent()).removeView(modalWindow);
```

...
...
but i'm not sure.

**Comment by shehabic** [comment]
Without knowing the implementation on how you're using it, I can't tell.

**Comment by javieramp** [comment]
Here is the implementation:

    private void generatePopupMenu() {
        if (hasAvailableProducts() && productIBTN != null) {
            DroppyMenuPopup.Builder droppyBuilder = new DroppyMenuPopup.Builder(this, productIBTN);

            for (Product product : getAvailableProductList()) {
                Drawable drawable = getDrawableSquare(product.getCode(), Color.WHITE, product.getColor());
                DroppyMenuItem droppyMenuItem = new DroppyMenuItem(product.getName());
                droppyMenuItem.setIcon(drawable);

                droppyBuilder.addMenuItem(droppyMenuItem)
                        .addSeparator();
            }

            droppyBuilder.setOnClick(new DroppyClickCallbackInterface() {
                @Override
                public void call(View v, int id) {
                    e("DroppyMenuPopup", "Clicked on: " + String.valueOf(id));

                    if (hasAvailableProducts()) {
                        setSelectedProduct(id);

                        MapFragment mapFragment = (MapFragment) findFragmentByTag(MapFragment.TAG);
                        if (mapFragment != null){
                            mapFragment.getLastKnowLocation();
                        }
                    }
                }
            });

            droppyBuilder.build();
        }
    }

**Comment by lukasberant** [comment]
I also have this problem...

**Comment by davidvavra** [comment]
+1 Every time I call dismiss() when popup is not opened, I get NullPointer

**Comment by shehabic** [comment]
Fixed in release 0.6.0
